staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -48,3 +48,47 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown `json:"configuration"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ConnectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService `json:"connectionServices"`
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// Name is the name of the connectivity service.
+	// Currently supported values are "Master" and "Cluster"
+	Name string `json:"name"`
+
+	// Connection is the exact information used to configure the network proxy
+	Connection Connection `json:"connection"`
+}
+
+// ConnectivityServiceConfiguration provides the configuration for a single network proxy client.
+type Connection struct {
+	// Type is the type of connection used to connect from client to network/proxy-server.
+	// Currently supported values are "http-connect" and "direct".
+	Type string `json:"type"`
+
+	// URL is the location of the proxy server to connect to.
+	// As an example it might be "https://127.0.0.1:8131"
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// CABundle is the file location of the CA to be used to determine trust with the network-proxy.
+	// +optional
+	CABundle string `json:"caBundle,omitempty"`
```

Comment:
Having `caBundle` and `clientCertFile` both point to a file is inconsistent.  I think the kubeconfig serialization is consistent https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/clientcmd/api/types.go#L89-L100.

Having an inline data option for the cabundle (not the cert/key pair below) is a thing that I like to avoid references.

Reply:
Switched the naming of these fields to be consistent with clientcmd.